pydantic/fields.py

```diff
@@ -36,16 +51,117 @@ class Validator(NamedTuple):
+    :param title: can be any string, used in the schema
+    :param description: can be any string, used in the schema
+    :param gt: only applies to numbers, requires the field to be "greater than". The schema
+      will have an ``exclusiveMinimum`` validation keyword
+    :param ge: only applies to numbers, requires the field to be "greater than or equal to". The
+    schema will have a ``minimum`` validation keyword
+    :param lt: only applies to numbers, requires the field to be "less than". The schema
+    will have an ``exclusiveMaximum`` validation keyword
+    :param le: only applies to numbers, requires the field to be "less than or equal to". The
+    schema will have a ``maximum`` validation keyword
+    :param min_length: only applies to strings, requires the field to have a minimum length. The
+    schema will have a ``maximum`` validation keyword
+    :param max_length: only applies to strings, requires the field to have a maximum length. The
+    schema will have a ``maxLength`` validation keyword
+    :param regex: only applies to strings, requires the field match agains a regular expression
+    pattern string. The schema will have a ``pattern`` validation keyword
+    :param **extra: any additional keyword arguments will be added as is to the schema
     """
 
-    __slots__ = 'default', 'alias', 'title', 'extra'
+    __slots__ = (
+        'default',
+        'alias',
+        'title',
+        'description',
+        'gt',
+        'ge',
+        'lt',
+        'le',
+        'min_length',
+        'max_length',
+        'regex',
+        'extra',
+    )
 
-    def __init__(self, default, *, alias=None, title=None, **extra):
+    def __init__(
+        self,
+        default,
+        *,
+        alias: str = None,
+        title: str = None,
+        description: str = None,
+        gt: float = None,
+        ge: float = None,
+        lt: float = None,
+        le: float = None,
+        min_length: int = None,
+        max_length: int = None,
+        regex: str = None,
+        **extra,
+    ):
         self.default = default
         self.alias = alias
         self.title = title
+        self.description = description
         self.extra = extra
+        self.gt = gt
+        self.ge = ge
+        self.lt = lt
+        self.le = le
+        self.min_length = min_length
+        self.max_length = max_length
+        self.regex = regex
+
+
+_numeric_types = (int, float, Decimal)
+_blacklist = (EmailStr, DSN, UrlStr, ConstrainedStr, ConstrainedInt, ConstrainedFloat, ConstrainedDecimal)
+_str_attrs = ('max_length', 'min_length', 'regex')
+_numeric_attrs = ('gt', 'lt', 'ge', 'le')
+_map_types_const = {str: constr, int: conint, float: confloat, Decimal: condecimal}
+
+
+def get_annotation_from_schema(annotation, schema):
+    """Get an annotation with validation implemented for numbers and strings based on the schema.
+
+    :param annotation: an annotation from a field specification, as ``str``, ``ConstrainedStr`` or the
```

Comment:
no need for ` or the return value of constr(max_length=10)` since that is a `ConstrainedStr`

Reply:
Sure. 

I had it that ways as I understand the return of `constr` would be of type `ConstrainedStrValue`, that inherits from `ConstranedStr`, because of the:

```Python
return type('ConstrainedStrValue', (ConstrainedStr,), namespace)
```

But I also think that it sounds difficult to understand and would be an overcomplication there in the docstings. So, if it's ok for you to just have up to the `ConstrainedStr`, then I'm going to remove that last part.

Reply:
ok